tests/pytests/test_case.py

```diff
@@ -0,0 +1,375 @@
+    env.assertEqual(categories['low'], 1)
+    env.assertEqual(categories['medium'], 2)
+    env.assertEqual(categories['high'], 1)
+
+def testNestedCaseAndGroup(env):
+    """Test nested case functions and group by"""
+    env.expect('FT.CREATE', 'idx_nested', 'SCHEMA', 't', 'TEXT').ok()
+    conn = getConnectionByEnv(env)
+
+    # Add documents with different text values
+    conn.execute_command('HSET', 'doc1', 't', 'pizza')
+    conn.execute_command('HSET', 'doc2', 't', 'pasta')
+    conn.execute_command('HSET', 'doc3', 't', 'burger')
+    conn.execute_command('HSET', 'doc4', 't', 'salad')
+
+    # Test nested case expressions
+    res = conn.execute_command(
+        'FT.AGGREGATE', 'idx_nested', '*',
+        'LOAD', '1', '@t',
+        'APPLY', 'case(contains(@t, "pizza"), ' \
+        '              "Italian", ' \
+        '              case(contains(@t, "pasta"), ' \
+        '                   "Italian", ' \
+        '                   case(contains(@t, "burger"), "American", "Other")))', 'AS', 'cuisine',
+        'GROUPBY', '1', '@cuisine',
+        'REDUCE', 'COUNT', '0', 'AS', 'count',
+        'SORTBY', '2', '@cuisine', 'ASC',
+        'DIALECT', '2')
+
+    # Expected: 3 groups - American, Italian, Other
+    env.assertEqual(res[0], 3)
+    cuisines = {row[1]: int(row[3]) for row in res[1:]}
+    env.assertEqual(cuisines['American'], 1)
+    env.assertEqual(cuisines['Italian'], 2)
+    env.assertEqual(cuisines['Other'], 1)
+
+def testCaseWithLogicalOperators(env):
+    """Test case function with logical operators (AND, OR, NOT)"""
+    conn = getConnectionByEnv(env)
+
+    # Create index with multiple fields
+    env.expect('FT.CREATE', 'idx_logic', 'SCHEMA',
+               'category', 'TAG',
+               'price', 'NUMERIC',
+               'in_stock', 'TAG').ok()
+
+    # Add test documents
+    conn.execute_command('HSET', 'prod1', 'category', 'electronics', 'price', '100', 'in_stock', 'yes')
+    conn.execute_command('HSET', 'prod2', 'category', 'electronics', 'price', '200', 'in_stock', 'no')
+    conn.execute_command('HSET', 'prod3', 'category', 'books', 'price', '15', 'in_stock', 'yes')
+    conn.execute_command('HSET', 'prod4', 'category', 'books', 'price', '25', 'in_stock', 'no')
+
+    # Test case with logical operators
+    res = conn.execute_command(
+        'FT.AGGREGATE', 'idx_logic', '*',
+        'LOAD', '3', '@category', '@price', '@in_stock',
+        'APPLY', 'case((@category == "electronics") && (@price < 150), "budget_electronics", case((@category == "books") || (@in_stock == "yes"), "available_items", "other"))', 'AS', 'product_class',
+        'GROUPBY', '1', '@product_class',
+        'REDUCE', 'COUNT', '0', 'AS', 'count',
+        'SORTBY', '2', '@product_class', 'ASC',
+        'DIALECT', '2')
+
+    # Expected results:
+    # - budget_electronics: prod1
+    # - available_items: prod2, prod3
+    # - other: prod4
+    # Total groups should be 3
```

Comment:
For complex conditions consider not using groupby
Expected seems to be 
prod1 first case true ==> budget_electronics
prod2 first case false, second case false ==> other
prod3 first case false, second case true ==> available_items
prod4 first case false, second case true ==> available_items

Reply:
OK. Tests modified.